test/test_torch.py

```diff
@@ -12828,6 +12855,179 @@ def test_function_unwrap_message(self):
+            # internal overlap check for inplace_op
+            self.binary_check_internal_mem_overlap(self, inplace_op,
+                                                   device=device)
+            # input/output overlap check for out op
+            self.binary_check_input_output_mem_overlap(self, out_op, device)
+
+    @torchtest.for_all_device_types()
+    def test_addXXX_mem_overlap(self, device):
+        ops = [
+            (torch.addcmul, torch.Tensor.addcmul_),
+            (torch.addcdiv, torch.Tensor.addcdiv_)
+        ]
+        for (out_op, inplace_op) in ops:
+            # internal overlap check for inplace_op
+            input = torch.randn(1, device=device).expand(3, 3)
+            tensor1 = torch.randn_like(input)
+            tensor2 = torch.randn_like(input)
+            with self.assertRaisesRegex(RuntimeError, 'single memory location'):
+                inplace_op(input, tensor1, tensor2)
+            # input/output overlap check for out op
+            sz = 3
+            data = torch.randn(sz * 4, device=device)
+
+            def _test(op, output, input, tensor1, tensor2):
+                output_exp = torch.empty_like(output)
+                op(input, tensor1, tensor2, out=output_exp)
+                self.assertEqual(op(input, tensor1, tensor2, out=output),
+                                 output_exp, op.__name__)
+
+            # output is identical to input, tensor1 and tensor2:
+            _test(out_op, output=data[0:sz], input=data[0:sz],
+                  tensor1=data[0:sz], tensor2=data[0:sz])
+            # output, input, tensor1 and tensor2 are independent:
+            _test(out_op, output=data[0:sz], input=data[sz:2 * sz],
+                  tensor1=data[2 * sz:3 * sz], tensor2=data[3 * sz:4 * sz])
+            # input, tensor1 and tensor2 have partial overlaps:
+            _test(out_op, output=data[0:sz], input=data[sz:2 * sz],
+                  tensor1=data[sz + 1:2 * sz + 1],
+                  tensor2=data[sz + 2:2 * sz + 2])
+            # output and input have a partial overlap:
+            with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
+                _test(out_op, output=data[0:sz], input=data[1:1 + sz],
+                      tensor1=data[2 * sz:3 * sz], tensor2=data[3 * sz:4 * sz])
+            # output and tensor1 have a partial overlap:
+            with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
+                _test(out_op, output=data[0:sz], input=data[2 * sz:3 * sz],
+                      tensor1=data[1:1 + sz], tensor2=data[3 * sz:4 * sz])
+            # output and tensor2 have a partial overlap:
+            with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
+                _test(out_op, output=data[0:sz], input=data[2 * sz:3 * sz],
+                      tensor1=data[3 * sz:4 * sz], tensor2=data[1:1 + sz])
+
+    @torchtest.for_all_device_types()
+    def test_copy_mem_overlap(self, device):
```

Comment:
Is it possible to just run `binary_check_internal_mem_overlap` and `test_unary_out_op_mem_overlap` on this?

Reply:
addressed